model/src/common/pagination/types.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Result of pagination containing page information
+ * @property {number} page - The current page number.
+ * @property {number} perPage - The number of items per page.
+ * @property {number} totalItems - The total number of items available.
+ * @property {number} totalPages - The total number of pages available.
+ */
+export interface PaginationResult {
+  page: number
+  perPage: number
+  totalItems: number
+  totalPages: number
+}
```

Comment:
PR looks ace

Would you mind moving the comments above the properties?

Just to make sure type tooltips get the descriptions coming through like this:

<img width="438" alt="Descriptions in tooltips" src="https://github.com/user-attachments/assets/527cd49e-a762-4174-a07a-3df306ca1ebd">

Reply:
Which I know is a pain, because they started life as JSDoc 😣

Would welcome a tech debt PR one day to migrate `model` to JSDoc (with generated types still)

Reply:
I always like to be as descriptive as possible but it can be a bit uneeded tbf.

Reply:
Ah sorry @mokhld yeah same

I meant like this. When the comment is above the property it appears in tooltips

```ts
/**
 * Result of pagination containing page information
 */
export interface PaginationResult {
  /**
   * The current page number.
   */
  page: number

  /**
   * The number of items per page.
   */
  perPage: number

  /**
   * The total number of items available.
   */
  totalItems: number

  /**
   * The total number of pages available.
   */
  totalPages: number
}
```

Reply:
ahh got you.